contributed_definitions/nyaml/NXpeak.yaml

```diff
@@ -1,59 +1,140 @@
       the peak represents/identifies.
-  peak_model:
+  lineshape:
     doc: |
       Is the peak described analytically via a functional form
       or is it empirically defined via measured/reported
-      intensity/counts as a function of an independent variable.
+      intensity/counts as a function of an independent variable?
       
-      If the functional form is not empirical or gaussian, users
-      should enter other for the peak_model and add relevant details
+      If the functional form is not empirical or Gaussians, users
+      should enter other for the description and add relevant details
       in the NXcollection.
-    enumeration: [empirical, gaussian, lorentzian, other]
+    enumeration: [empirical, gaussian, lorentzian, voigt, other]
+  lineshape_other:
+    doc: |
+      Human-readable descritpion of the lineshape used for the peak if it does
+      not fit into the enumeration given in `lineshape`.
   position(NX_NUMBER):
     unit: NX_ANY
     doc: |
       In the case of an empirical description of the peak and its shoulders,
       this array holds the position values for the independent variable.
-    dimensions:
-      rank: 1
-      dim: [[1, n_support]]
+    dim: (n_support,)
   intensity(NX_NUMBER):
     unit: NX_ANY
     doc: |
       In the case of an empirical description of the peak and its shoulders,
       this array holds the intensity/count values at each position.
-    dimensions:
-      rank: 1
-      dim: [[1, n_support]]
-  (NXcollection):
+    dim: (n_support,)
+  width(NX_NUMBER):
+    unit: NX_ANY
+    doc: |
+      Width of a peak at a defined fraction of the peak height.
+      
+      Usually, this will be the Full Width at Half Maximum of the peak (FWHM).
+      For asymmetric peaks, convenient measures of peak width are the half-widths of
+      each side of the peak at half maximum intensity.
+    \@type:
+      doc: |
+        Specification of the definition of `width`.
+      enumeration:
+        fwhm:
+          doc: |
+            Full Width at Half Maximum of the peak.
+        hwhm:
+          doc: |
+            Half Width at Half Maximum of the peak.
+        half_widths_asymmetric:
+          doc: |
+            For asymmetrical peaks, the half-widths of each side of the peak at
+            half maximum intensity is given.
+        other:
+          doc: |
+            For other peak width definitions.
+  functional_form:
     doc: |
-      In the case of an analytical description (or if peak_model is other) this
+      A formula to describe the line shape using set of parameters as entered
+      by the `peak_parameters` fields.
+  peak_parameters(NXparameters):
+    doc: |
+      In the case of an analytical description (or if description is other) this
       collection holds parameter of (and eventually) the functional form.
       For example in the case of Gaussians mu, sigma, cut-off values,
       and background intensity are relevant parameter.
```

Comment:
Just the parameter, not the function form, no? Not sure I know NXparameters well enough.
Background is rather for an instance of NXfit_background, no?

Reply:
Correct